shared/libraries/config_protocol/tests/test_core_events.cpp

```diff
@@ -31,18 +32,17 @@ class ConfigCoreEventTest : public testing::Test
         server = std::make_unique<ConfigProtocolServer>(serverDevice, std::bind(&ConfigCoreEventTest::serverNotificationReady, this, std::placeholders::_1));
 
         clientContext = NullContext();
-        client = std::make_unique<ConfigProtocolClient>(clientContext, std::bind(&ConfigCoreEventTest::sendRequest, this, std::placeholders::_1), nullptr);
+        client = std::make_unique<ConfigProtocolClient<ConfigClientDeviceImpl>>(clientContext, std::bind(&ConfigCoreEventTest::sendRequest, this, std::placeholders::_1), nullptr);
 
-        client->connect();
-        client->getDevice().asPtr<IPropertyObjectInternal>().enableCoreEventTrigger();
-        clientDevice = client->getDevice();
+        clientDevice = client->connect();
+        clientDevice.asPtr<IPropertyObjectInternal>().enableCoreEventTrigger();
     }
 
 protected:
     DevicePtr serverDevice;
     DevicePtr clientDevice;
     std::unique_ptr<ConfigProtocolServer> server;
-    std::unique_ptr<ConfigProtocolClient> client;
+    std::unique_ptr<ConfigProtocolClient<ConfigClientDeviceImpl>> client;
     ContextPtr clientContext;
     BaseObjectPtr notificationObj;
```

Comment:
General comment - it'd be great if these tests could be simply used with the a different transport layer protocol. So we'd run the same test, but with different server/client callbacks.

It'd be a waste to not reuse these tests.